subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -405,9 +735,250 @@ int fp_storage_ak_save(const struct fp_account_key *account_key)
 			"order may change at reboot.");
 	}
 
+	if (IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND) && ak_overwritten) {
+		/* Account Key overwritten. Remove bonds releted with overwritten Account Key. */
+		bonds_with_invalid_id_remove_completely();
+	}
+
+	return 0;
+}
+
+static int ak_id_get(uint8_t *id, const struct fp_account_key *account_key)
+{
+	for (size_t i = 0; i < account_key_count; i++) {
+		if (!memcmp(account_key_list[i].key, account_key->key, FP_ACCOUNT_KEY_LEN)) {
+			*id = ACCOUNT_KEY_METADATA_FIELD_GET(account_key_metadata[i], ID);
+			return 0;
+		}
+	}
+
+	return -ESRCH;
+}
+
+static struct fp_bond *fp_bond_free_get(void)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(fp_bonds); i++) {
+		if (bt_addr_le_eq(&fp_bonds[i].nv.addr, BT_ADDR_LE_ANY)) {
+			return &fp_bonds[i];
+		}
+	}
+
+	return NULL;
+}
+
+static struct fp_bond *bonded_duplicate_get(const void *conn_ctx, const bt_addr_le_t *addr)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(fp_bonds); i++) {
+		if ((fp_bonds[i].conn_ctx != conn_ctx) &&
+		    fp_bonds[i].bonded &&
+		    bt_addr_le_eq(&fp_bonds[i].nv.addr, addr)) {
+			return &fp_bonds[i];
+		}
+	}
+
+	return NULL;
+}
+
+static void bond_update_addr(const void *conn_ctx, const bt_addr_le_t *new_addr)
+{
+	int err;
+	struct fp_bond *bond;
+	struct fp_bond bond_rollback_copy;
+
+	bond = fp_bond_get(conn_ctx);
+	if (!bond) {
+		return;
+	}
+	bond_rollback_copy = *bond;
+
+	bond->nv.addr = *new_addr;
+
+	err = fp_bond_settings_save(bond, &bond_rollback_copy);
+	if (err) {
+		return;
+	}
+}
+
+static struct fp_bond *bonded_by_addr_get(const bt_addr_le_t *addr)
```

Comment:
Group the getter functions together and name to indicate the similarity

fp_bond_info_by_addr_get(...
fp_bond_info_by_conn_get(...
fp_bond_info_duplicate_get(...)
fp_bond_info_free_get(...)

Reply:
Or maybe even `xyz_to_fp_bond` for the conversions?

Reply:
I'll stick with Kamil's approach